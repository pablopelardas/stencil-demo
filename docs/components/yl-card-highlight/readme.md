# yl-card-highlight


<!-- Auto Generated Below -->


## Properties

| Property      | Attribute           | Description                                                                            | Type     | Default                         |
| ------------- | ------------------- | -------------------------------------------------------------------------------------- | -------- | ------------------------------- |
| `headerLevel` | `data-header-level` | El nivel de <h?></h?> que se va a generar                                              | `number` | `2`                             |
| `image`       | `data-image`        | La imagen y sus variantes para cada vista. Es un objeto que viene en un encodedUri     | `string` | `encodeURI(JSON.stringify({}))` |
| `links`       | `data-links`        | Los links que se van a mostrar en la carpeta. Es un arreglo que viene en un encodedUri | `string` | `encodeURI(JSON.stringify([]))` |
| `position`    | `data-position`     | La posiciÃ³n de la imagen, pueder ser top \| bottom \| left \| right                    | `string` | `'top'`                         |


## Methods

### `exportHtml() => Promise<string>`

Exporta el componente a HTML para poder ser usado en cualquier lado con las mismas propiedades

#### Returns

Type: `Promise<string>`

Retorna el HTML del componente

### `manifest() => Promise<object>`

Exporta el manifest del componente para poder ser usado en el builder, devuelve un objeto con la siguiente estructura
{
name: string, (Nombre del componente)
title: string, (Titulo del componente)
preview: string, (url de la imagen)
category: string, (General, Formularios, etc)
initHTML: string (HTML minimo del componente)

#### Returns

Type: `Promise<object>`

Retorna el manifest del componente para poder ser usado en el builder


## Slots

| Slot            | Description             |
| --------------- | ----------------------- |
| `"description"` | La descripcion del card |
| `"title"`       | El titulo del card      |


## Shadow Parts

| Part            | Description                                         |
| --------------- | --------------------------------------------------- |
| `"description"` | Part de la descripcion para poder cambiar el estilo |
| `"link"`        | Part del link para poder cambiar el estilo          |
| `"title"`       | Part del titulo para poder cambiar el estilo        |


## CSS Custom Properties

| Name                                  | Description                        |
| ------------------------------------- | ---------------------------------- |
| `--yl-card-highlight-big-font-size`   | font size for the title.           |
| `--yl-card-highlight-font-family`     | font family used in the component. |
| `--yl-card-highlight-max-width`       | max width host element can reach.  |
| `--yl-card-highlight-small-font-size` | font size for the description.     |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
