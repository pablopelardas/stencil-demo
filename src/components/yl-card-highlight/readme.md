# yl-card-highlight


<!-- Auto Generated Below -->


## Properties

| Property      | Attribute           | Description | Type     | Default     |
| ------------- | ------------------- | ----------- | -------- | ----------- |
| `headerLevel` | `data-header-level` |             | `number` | `undefined` |
| `image`       | `data-image`        |             | `string` | `undefined` |
| `links`       | `data-links`        |             | `string` | `undefined` |
| `position`    | `data-position`     |             | `string` | `undefined` |


## Shadow Parts

| Part            | Description |
| --------------- | ----------- |
| `"description"` |             |
| `"link"`        |             |


## CSS Custom Properties

| Name                                  | Description                        |
| ------------------------------------- | ---------------------------------- |
| `--yl-card-highlight-big-font-size`   | font size for the title.           |
| `--yl-card-highlight-font-family`     | font family used in the component. |
| `--yl-card-highlight-max-width`       | max width host element can reach.  |
| `--yl-card-highlight-small-font-size` | font size for the description.     |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
