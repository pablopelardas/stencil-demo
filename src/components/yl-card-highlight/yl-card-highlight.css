:host {
  box-sizing: border-box;
  display: flex;
  /**
    * @prop --yl-card-highlight-max-width: max width host element can reach.
    */
  --yl-card-highlight-max-width: 500px;
  /**
    * @prop --yl-card-highlight-font-family: font family used in the component.
    */
  --yl-card-highlight-font-family: var(--yl-theme-font-family);
  /**
    * @prop --yl-card-highlight-big-font-size: font size for the title.
    */
  --yl-card-highlight-big-font-size: 24px;
  /**
    * @prop --yl-card-highlight-small-font-size: font size for the description.
    */
  --yl-card-highlight-small-font-size: 16px;

  border: 1px solid #eaeaea;
  max-width: var(--yl-card-highlight-max-width);
  padding: 20px;
}

:host(.bottom) {
  flex-direction: column-reverse;
}

:host(.top) {
  flex-direction: column;
}

:host(.left) {
  flex-direction: row;
}

:host(.right) {
  flex-direction: row-reverse;
}

:host(.right),
:host(.left) {
  gap: 20px;
}

:host(.right)>div,
:host(.left)>div {
  width: 50%;
}

:host(.top)>div,
:host(.bottom)>div {
  width: 100%;
}

.cover {
  position: relative;
  z-index: 1;
  overflow: hidden;
}

.cover img {
  width: 100%;
  object-fit: cover;
  display: block;
}

.cover div {
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  padding: 10px 10px 15px;
  z-index: 2;
  background: rgba(0, 0, 0, 0.7);
  color: #fff;
  font-size: 13px;
  font-weight: bold;
  line-height: 18px;
  margin-bottom: 0px;
  opacity: 0;
  transition: opacity 0.3s ease-in;
}

.cover:hover div {
  opacity: 1;
}

.text {
  display: flex;
  flex-direction: column;
  justify-content: center;

}

.title {
  text-transform: uppercase;
  font-size: var(--yl-card-highlight-big-font-size);
  font-weight: 500;
  font-family: var(--yl-card-highlight-font-family);
  margin-bottom: 0px;
}

.description {
  font-size: var(--yl-card-highlight-small-font-size);
  font-family: var(--yl-card-highlight-font-family);
  margin-bottom: 20px;
}

ul {
  list-style: none;
  padding: 0;
  margin: 0;
}

ul li {
  display: flex;
  align-items: center;
  margin-bottom: 10px;
}

ul li a {
  color: #808080;
  text-decoration: none;
  font-weight: bold;
  font-size: var(--yl-card-highlight-small-font-size);
  font-family: var(--yl-card-highlight-font-family);
  text-transform: uppercase;
}

ul li a:hover {
  color: #393939;
}