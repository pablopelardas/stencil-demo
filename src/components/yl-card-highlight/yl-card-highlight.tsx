import { Component, Host, h, Prop, State, Listen, Method, Element } from '@stencil/core';
import { toAbsoluteUrl } from '../../utils/utils';

type ImageData = {
  src?: string;
  alt?: string;
  desktop?: string;
  tablet?: string;
  mobile?: string;
}

type Links = {
  href: string;
  label: string;
  title: string;
}

/**
 * @slot title - El titulo del card
 * @slot description - La descripcion del card
 * @part title - Part del titulo para poder cambiar el estilo
 * @part description - Part de la descripcion para poder cambiar el estilo
 * @part link - Part del link para poder cambiar el estilo
 */

@Component({
  tag: 'yl-card-highlight',
  styleUrl: 'yl-card-highlight.css',
  shadow: true,
})
export class YlCardHighlight {

  private imageData: ImageData;
  private linksData: Links[];

  @Element() el: HTMLElement;

  @State() viewport: string = 'desktop';
  @Listen('resize', { target: 'window' })
  handleResize() {
    this.updateViewport();
  }

  /** El nivel de <h?></h?> que se va a generar */
  @Prop({ attribute: 'data-header-level' }) headerLevel: number = 2;
  /** La imagen y sus variantes para cada vista. Es un objeto que viene en un encodedUri */
  @Prop({ attribute: 'data-image', }) image: string = encodeURI(JSON.stringify({}));
  /** Los links que se van a mostrar en la carpeta. Es un arreglo que viene en un encodedUri */
  @Prop({ attribute: 'data-links', }) links: string = encodeURI(JSON.stringify([]));
  /** La posici√≥n de la imagen, pueder ser top | bottom | left | right */
  @Prop({ attribute: 'data-position', }) position: string = 'top';

  componentWillLoad() {
    this.imageData = this.image ? JSON.parse(this.decodeUri(this.image)) : {};
    let { src, desktop, tablet, mobile } = this.imageData;
    if (!src && !desktop && !tablet && !mobile) {
      this.imageData = {
        src: 'https://via.placeholder.com/300x300.png?text=Imagen+no+encontrada',
        alt: 'Imagen no encontrada',
        desktop: 'https://via.placeholder.com/300x300.png?text=Imagen+no+encontrada',
        tablet: 'https://via.placeholder.com/300x300.png?text=Imagen+no+encontrada',
        mobile: 'https://via.placeholder.com/300x300.png?text=Imagen+no+encontrada',
      }
    }
    this.linksData = this.links ? JSON.parse(this.decodeUri(this.links)) : [];
    this.updateViewport();
  }

  /**
   * Exporta el componente a HTML para poder ser usado en cualquier lado con las mismas propiedades
   * @returns Retorna el HTML del componente
   */
  @Method() async exportHtml(): Promise<string> {
    const titleSlot = this.el.querySelector('div[slot="title"]').textContent;
    const descriptionSlot = this.el.querySelector('div[slot="description"]').textContent;
    console.log(titleSlot)
    return `
      <yl-card-highlight
        data-header-level="${this.headerLevel}"
        data-position="${this.position}"
        data-image="${this.image}"
        data-links="${this.links}"
      >
        <div slot="title">${titleSlot}</div>
        <div slot="description">${descriptionSlot}</div>
      </yl-card-highlight>
    `;
  }

  /**
   * Exporta el manifest del componente para poder ser usado en el builder, devuelve un objeto con la siguiente estructura
   * {
   * name: string, (Nombre del componente)
   * title: string, (Titulo del componente)
   * preview: string, (url de la imagen)
   * category: string, (General, Formularios, etc)
   * initHTML: string (HTML minimo del componente)
   * @returns Retorna el manifest del componente para poder ser usado en el builder
   */
  @Method() async manifest(): Promise<object> {
    return {
      name: 'yl-card-highlight',
      title: 'Tarjeta Destacada',
      preview: 'https://via.placeholder.com/300x300.png?text=Imagen+no+encontrada',
      category: 'General',
      initHTML: `
        <yl-card-highlight
          data-header-level,
          data-position,
          data-image,
          data-links
        >
          <div slot="title">Titulo</div>
          <div slot="description">Descripcion</div>
        </yl-card-highlight>
        `
    }
  }


  render() {
    return (
      <Host
        class={{
          [this.position]: true,
        }}
      >
        <div class="cover">
          <img
            src={this.imageSrc(this.getImageSrc(this.imageData, this.viewport))}
            alt={this.imageData.alt}
          />
          <div>
            <span>{this.imageData.alt}</span>
          </div>
        </div>
        <div class="text">
          <div innerHTML={this.getHeader(this.headerLevel)}></div>
          <p class="description" part='description'><slot name='description'>Descripcion</slot></p>
          <ul>
            {this.linksData.map((link) => (
              <li class='link-item' part='link'>
                <a href={link.href} title={link.title}>{link.label}</a>
              </li>
            ))}
          </ul>
        </div>
      </Host>
    );
  }

  private decodeUri(uri: string): string {
    return decodeURIComponent(uri);
  }

  private imageSrc(src: string): string {
    return toAbsoluteUrl(src);
  }

  private updateViewport() {
    const mobileQuery = window.matchMedia('(max-width: 768px)');
    const tabletQuery = window.matchMedia('(max-width: 1024px)');

    if (mobileQuery.matches) {
      this.viewport = 'mobile';
    } else if (tabletQuery.matches) {
      this.viewport = 'tablet';
    } else {
      this.viewport = 'desktop';
    }
  }

  private getImageSrc(imageData: ImageData, viewport: string): string {
    const { src, desktop, tablet, mobile } = imageData;
    if (!src && !desktop && !tablet && !mobile) return '';
    let imgSrc = imageData[viewport];
    if (!imgSrc) {
      if (viewport === 'mobile') {
        imgSrc = imageData.tablet || imageData.desktop || imageData.src;
      } else if (viewport === 'tablet') {
        imgSrc = imageData.desktop || imageData.src;
      } else {
        imgSrc = imageData.src;
      }
    }
    return imgSrc;
  }

  private getHeader(headerLevel: number = 2): string {
    if (headerLevel > 6) headerLevel = 6;
    return `<h${headerLevel} class=title part="title"><slot name='title'>Titulo</slot></h${headerLevel}>`;
  }
}