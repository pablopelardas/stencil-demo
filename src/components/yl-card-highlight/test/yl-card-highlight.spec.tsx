import { newSpecPage } from '@stencil/core/testing';
import { YlCardHighlight } from '../yl-card-highlight';

describe('yl-card-highlight internal logic', () => {
  it('builds', () => {
    expect(new YlCardHighlight()).toBeTruthy();
  })

  it('has default values', async () => {
    expect(new YlCardHighlight()).toEqual({
      headerLevel: 2,
      position: 'top',
      image: encodeURIComponent(JSON.stringify({})),
      links: encodeURIComponent(JSON.stringify([])),
      viewport: 'desktop',
    })
  })

  describe('decodeUri()', () => {
    it('decodes a URI-encoded string', () => {
      const component = new YlCardHighlight();
      const uri = 'https%3A%2F%2Fexample.com%2Fimage.jpg';
      const result = component['decodeUri'](uri);
      expect(result).toEqual('https://example.com/image.jpg');
    });
    it ('decodes an image object URI-encoded string', () => {
      const component = new YlCardHighlight();
      const uri = encodeURIComponent(JSON.stringify({
        src: 'https://example.com/image.jpg',
        alt: 'alt',
        desktop: 'https://example.com/image-desktop.jpg',
        tablet: 'https://example.com/image-tablet.jpg',
        mobile: 'https://example.com/image-mobile.jpg',
      }));
      const result = component['decodeUri'](uri);
      expect(result).toEqual(JSON.stringify({
        src: 'https://example.com/image.jpg',
        alt: 'alt',
        desktop: 'https://example.com/image-desktop.jpg',
        tablet: 'https://example.com/image-tablet.jpg',
        mobile: 'https://example.com/image-mobile.jpg',
      }));
    });
  });

  describe('getImageSrc()', () => {

    it('returns src when viewport is desktop', () => {
      const component = new YlCardHighlight();
      const imageData = {
        src: 'https://example.com/image.jpg',
        desktop: 'https://example.com/image-desktop.jpg',
      };
      const viewport = 'desktop';
      const result = component['getImageSrc'](imageData, viewport);
      expect(result).toEqual(imageData.desktop);
    });

    it('returns tablet when viewport is tablet', () => {
      const component = new YlCardHighlight();
      const imageData = {
        src: 'https://example.com/image.jpg',
        desktop: 'https://example.com/image-desktop.jpg',
        tablet: 'https://example.com/image-tablet.jpg',
      };
      const viewport = 'tablet';
      const result = component['getImageSrc'](imageData, viewport);
      expect(result).toEqual(imageData.tablet);
    });

    it('returns mobile when viewport is mobile', () => {
      const component = new YlCardHighlight();
      const imageData = {
        src: 'https://example.com/image.jpg',
        desktop: 'https://example.com/image-desktop.jpg',
        tablet: 'https://example.com/image-tablet.jpg',
        mobile: 'https://example.com/image-mobile.jpg',
      };
      const viewport = 'mobile';
      const result = component['getImageSrc'](imageData, viewport);
      expect(result).toEqual(imageData.mobile);
    });

    it('returns desktop when viewport is mobile and tablet is empty', () => {
      const component = new YlCardHighlight();
      const imageData = {
        src: 'https://example.com/image.jpg',
        desktop: 'https://example.com/image-desktop.jpg',
      };
      const viewport = 'mobile';
      const result = component['getImageSrc'](imageData, viewport);
      expect(result).toEqual(imageData.desktop);
    });

    it('returns tablet when viewport is tablet', () => {
      const component = new YlCardHighlight();
      const imageData = {
        src: 'https://example.com/image.jpg',
        desktop: 'https://example.com/image-desktop.jpg',
        tablet: 'https://example.com/image-tablet.jpg',
      };
      const viewport = 'tablet';
      const result = component['getImageSrc'](imageData, viewport);
      expect(result).toEqual(imageData.tablet);
    });

    it('returns src when viewport is mobile and tablet and desktop are empty', () => {
      const component = new YlCardHighlight();
      const imageData = {
        src: 'https://example.com/image.jpg',
      };
      const viewport = 'mobile';
      const result = component['getImageSrc'](imageData, viewport);
      expect(result).toEqual(imageData.src);
    });
  });

  describe('getHeader()', () => {
    it('returns h2 by default', () => {
      const component = new YlCardHighlight();
      const headerLevel = undefined;
      const result = component['getHeader'](headerLevel);
      expect(result).toEqual('<h2 class=title part="title"><slot name=\'title\'>Titulo</slot></h2>');
    });

    it('returns h1 when headerLevel is 1', () => {
      const component = new YlCardHighlight();
      const headerLevel = 1;
      const result = component['getHeader'](headerLevel);
      expect(result).toEqual('<h1 class=title part="title"><slot name=\'title\'>Titulo</slot></h1>');
    });

    it('returns h2 when headerLevel is 2', () => {
      const component = new YlCardHighlight();
      const headerLevel = 2;
      const result = component['getHeader'](headerLevel);
      expect(result).toEqual('<h2 class=title part="title"><slot name=\'title\'>Titulo</slot></h2>');
    });

    it('returns h3 when headerLevel is 3', () => {
      const component = new YlCardHighlight();
      const headerLevel = 3;
      const result = component['getHeader'](headerLevel);
      expect(result).toEqual('<h3 class=title part="title"><slot name=\'title\'>Titulo</slot></h3>');
    });
    it('returns h6 when headerLevel is bigger than 6', () => {
      const component = new YlCardHighlight();
      const headerLevel = 12;
      const result = component['getHeader'](headerLevel);
      expect(result).toEqual('<h6 class=title part="title"><slot name=\'title\'>Titulo</slot></h6>');
    });
  });

  describe('updateViewport()', () => {
    it('sets viewport to mobile when window width is less than 768px', () => {
      const component = new YlCardHighlight();
      const mobileQuery = {
        matches: true,
      };
      const tabletQuery = {
        matches: false,
      };
      window.matchMedia = jest.fn().mockImplementation((query) => {
        if (query === '(max-width: 768px)') {
          return mobileQuery;
        } else if (query === '(max-width: 1024px)') {
          return tabletQuery;
        }
      });
      component['updateViewport']();
      expect(component['viewport']).toEqual('mobile');
    });

    it('sets viewport to tablet when window width is between 768px and 1024px', () => {
      const component = new YlCardHighlight();
      const mobileQuery = {
        matches: false,
      };
      const tabletQuery = {
        matches: true,
      };
      window.matchMedia = jest.fn().mockImplementation((query) => {
        if (query === '(max-width: 768px)') {
          return mobileQuery;
        } else if (query === '(max-width: 1024px)') {
          return tabletQuery;
        }
      });
      component['updateViewport']();
      expect(component['viewport']).toEqual('tablet');
    });

    it('sets viewport to desktop when window width is greater than 1024px', () => {
      const component = new YlCardHighlight();
      const mobileQuery = {
        matches: false,
      };
      const tabletQuery = {
        matches: false,
      };
      window.matchMedia = jest.fn().mockImplementation((query) => {
        if (query === '(max-width: 768px)') {
          return mobileQuery;
        } else if (query === '(max-width: 1024px)') {
          return tabletQuery;
        }
      });
      component['updateViewport']();
      expect(component['viewport']).toEqual('desktop');
    });
  });
});

describe.only('yl-card-highlight rendering', () => {
  it('renders correctly with default props', async () => {
    const page = await newSpecPage({
      components: [YlCardHighlight],
      html: '<yl-card-highlight></yl-card-highlight>',
    });
    expect(page.root).toMatchSnapshot();
  });

  it('renders correctly with custom props', async () => {
    const page = await newSpecPage({
      components: [YlCardHighlight],
      html: `
      <yl-card-highlight data-header-level="1" data-position="top"
      data-image="%7B%22src%22%3A%22https%3A%2F%2Fth.bing.com%2Fth%2Fid%2FOIP.SMAmS1MxCpssY-raDGzy1wHaE7%3Fpid%3DImgDet%26rs%3D1%22%2C%22alt%22%3A%22imagen%20decorativa%22%7D"
      data-links="%5B%7B%22href%22%3A%22https%3A%2F%2Fapi.spotify.com%2Fv1%2Fartists%2F0OdUWJ0sBjDrqHygGUXeCF%22%2C%22label%22%3A%22Wilco%22%2C%22title%22%3A%22Wilco%22%7D%2C%7B%22href%22%3A%22https%3A%2F%2Fapi.spotify.com%2Fv1%2Fartists%2F7jy3rLJdDQY21OgRLCZ9sD%22%2C%22label%22%3A%22Tomas%22%2C%22title%22%3A%22Tomas%22%7D%5D">
      </yl-card-highlight>
      `,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('renders correctly with custom props and slot', async () => {
    const page = await newSpecPage({
      components: [YlCardHighlight],
      html: `
      <yl-card-highlight data-header-level="1" data-position="top"
    data-image="%7B%22src%22%3A%22https%3A%2F%2Fth.bing.com%2Fth%2Fid%2FOIP.SMAmS1MxCpssY-raDGzy1wHaE7%3Fpid%3DImgDet%26rs%3D1%22%2C%22alt%22%3A%22imagen%20decorativa%22%7D"
    data-links="%5B%7B%22href%22%3A%22https%3A%2F%2Fapi.spotify.com%2Fv1%2Fartists%2F0OdUWJ0sBjDrqHygGUXeCF%22%2C%22label%22%3A%22Wilco%22%2C%22title%22%3A%22Wilco%22%7D%2C%7B%22href%22%3A%22https%3A%2F%2Fapi.spotify.com%2Fv1%2Fartists%2F7jy3rLJdDQY21OgRLCZ9sD%22%2C%22label%22%3A%22Tomas%22%2C%22title%22%3A%22Tomas%22%7D%5D">
    <div slot=title>Header desde Slot</div>
    <div slot=description>Descripci√≥n desde Slot</div>
  </yl-card-highlight>
      `,
    })
    expect(page.root).toMatchSnapshot();
  });
});
