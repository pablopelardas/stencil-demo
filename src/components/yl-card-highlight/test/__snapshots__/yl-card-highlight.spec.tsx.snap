// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`yl-card-highlight rendering renders correctly with custom props 1`] = `
<yl-card-highlight class="top" data-header-level="1" data-image="%7B%22src%22%3A%22https%3A%2F%2Fth.bing.com%2Fth%2Fid%2FOIP.SMAmS1MxCpssY-raDGzy1wHaE7%3Fpid%3DImgDet%26rs%3D1%22%2C%22alt%22%3A%22imagen%20decorativa%22%7D" data-links="%5B%7B%22href%22%3A%22https%3A%2F%2Fapi.spotify.com%2Fv1%2Fartists%2F0OdUWJ0sBjDrqHygGUXeCF%22%2C%22label%22%3A%22Wilco%22%2C%22title%22%3A%22Wilco%22%7D%2C%7B%22href%22%3A%22https%3A%2F%2Fapi.spotify.com%2Fv1%2Fartists%2F7jy3rLJdDQY21OgRLCZ9sD%22%2C%22label%22%3A%22Tomas%22%2C%22title%22%3A%22Tomas%22%7D%5D" data-position="top">
  <mock:shadow-root>
    <div class="cover">
      <img alt="imagen decorativa" src="https://th.bing.com/th/id/OIP.SMAmS1MxCpssY-raDGzy1wHaE7?pid=ImgDet&amp;rs=1">
      <div>
        <span>
          imagen decorativa
        </span>
      </div>
    </div>
    <div class="text">
      <div>
        <h1 class="title" part="title">
          <slot name="title">
            Titulo
          </slot>
        </h1>
      </div>
      <p class="description" part="description">
        <slot name="description">
          Descripcion
        </slot>
      </p>
      <ul>
        <li class="link-item" part="link">
          <a href="https://api.spotify.com/v1/artists/0OdUWJ0sBjDrqHygGUXeCF" title="Wilco">
            Wilco
          </a>
        </li>
        <li class="link-item" part="link">
          <a href="https://api.spotify.com/v1/artists/7jy3rLJdDQY21OgRLCZ9sD" title="Tomas">
            Tomas
          </a>
        </li>
      </ul>
    </div>
  </mock:shadow-root>
</yl-card-highlight>
`;

exports[`yl-card-highlight rendering renders correctly with custom props and slot 1`] = `
<yl-card-highlight class="top" data-header-level="1" data-image="%7B%22src%22%3A%22https%3A%2F%2Fth.bing.com%2Fth%2Fid%2FOIP.SMAmS1MxCpssY-raDGzy1wHaE7%3Fpid%3DImgDet%26rs%3D1%22%2C%22alt%22%3A%22imagen%20decorativa%22%7D" data-links="%5B%7B%22href%22%3A%22https%3A%2F%2Fapi.spotify.com%2Fv1%2Fartists%2F0OdUWJ0sBjDrqHygGUXeCF%22%2C%22label%22%3A%22Wilco%22%2C%22title%22%3A%22Wilco%22%7D%2C%7B%22href%22%3A%22https%3A%2F%2Fapi.spotify.com%2Fv1%2Fartists%2F7jy3rLJdDQY21OgRLCZ9sD%22%2C%22label%22%3A%22Tomas%22%2C%22title%22%3A%22Tomas%22%7D%5D" data-position="top">
  <mock:shadow-root>
    <div class="cover">
      <img alt="imagen decorativa" src="https://th.bing.com/th/id/OIP.SMAmS1MxCpssY-raDGzy1wHaE7?pid=ImgDet&amp;rs=1">
      <div>
        <span>
          imagen decorativa
        </span>
      </div>
    </div>
    <div class="text">
      <div>
        <h1 class="title" part="title">
          <slot name="title">
            Titulo
          </slot>
        </h1>
      </div>
      <p class="description" part="description">
        <slot name="description">
          Descripcion
        </slot>
      </p>
      <ul>
        <li class="link-item" part="link">
          <a href="https://api.spotify.com/v1/artists/0OdUWJ0sBjDrqHygGUXeCF" title="Wilco">
            Wilco
          </a>
        </li>
        <li class="link-item" part="link">
          <a href="https://api.spotify.com/v1/artists/7jy3rLJdDQY21OgRLCZ9sD" title="Tomas">
            Tomas
          </a>
        </li>
      </ul>
    </div>
  </mock:shadow-root>
  <div slot="title">
    Header desde Slot
  </div>
  <div slot="description">
    Descripci√≥n desde Slot
  </div>
</yl-card-highlight>
`;

exports[`yl-card-highlight rendering renders correctly with default props 1`] = `
<yl-card-highlight class="top">
  <mock:shadow-root>
    <div class="cover">
      <img alt="Imagen no encontrada" src="https://via.placeholder.com/300x300.png?text=Imagen+no+encontrada">
      <div>
        <span>
          Imagen no encontrada
        </span>
      </div>
    </div>
    <div class="text">
      <div>
        <h2 class="title" part="title">
          <slot name="title">
            Titulo
          </slot>
        </h2>
      </div>
      <p class="description" part="description">
        <slot name="description">
          Descripcion
        </slot>
      </p>
      <ul></ul>
    </div>
  </mock:shadow-root>
</yl-card-highlight>
`;
